### PR TITLE
Get Dictionary instead of GrokResult with AsDictionary method (#72)

### PR DESCRIPTION
**Pull request by GioMaz**
```cs
Grok grok = new Grok("%{MONTHDAY:month}-%{MONTHDAY:day}-%{MONTHDAY:year} %{TIME:timestamp};%{WORD:id};%{LOGLEVEL:loglevel};%{WORD:func};%{GREEDYDATA:msg}");

string logs = @"06-21-19 21:00:13:589241;15;INFO;main;DECODED: 775233900043 DECODED BY: 18500738 DISTANCE: 1.5165
                06-22-19 22:00:13:589265;156;WARN;main;DECODED: 775233900043 EMPTY DISTANCE: --------";

var grokResult = grok.AsDictionary(logs);

foreach (var item in grokResult)
{
    Console.WriteLine($"{item.Key}:");
    foreach (var value in item.Value)
    {
        Console.WriteLine("\t" + value);
    }
}
```

this will now print:

```
month:
        06
        06
day:
        21
        22
year:
        19
        19
timestamp:
        21:00:13:589241
        22:00:13:589265
id:
        15
        156
loglevel:
        INFO
        WARN
func:
        main
        main
msg:
        DECODED: 775233900043 DECODED BY: 18500738 DISTANCE: 1.5165
        DECODED: 775233900043 EMPTY DISTANCE: --------
```

### CODE REVIEW

File: src/Grok.Net/Grok.cs [hunk 1]
```diff
@@ -86,6 +86,41 @@ public GrokResult Parse(string text)
             return new GrokResult(grokItems);
         }
 
+        public Dictionary<string, List<object>> AsDictionary(string text)
```
Comment (Marusyk) [inline src/Grok.Net/Grok.cs:89]: This "additional" representation of the results is definitely NOT the responsibility of this class.
What if tomorrow we need another view? This approach violates the S and O from SOLID.

**Review by Marusyk — CHANGES_REQUESTED** [review]
Thanks for your time. 
The changes seem overcomplicated and violate some good OOP practices. I'm sure we can done this in "1 line of code"  
tip - look at the title of the #72 😉

File: src/Grok.Net/Grok.cs (hunk 1 above)
Comment (ricardSiliuk) [inline src/Grok.Net/Grok.cs:89]: AFAIK this violates dotnet convention of `AsX()` functions not allocating though I can't find a definitive source on this right now :/ Extension method of `GrokResult.ToDictionary()` might be more correct?

Comment (GioMaz) [reply src/Grok.Net/Grok.cs:89]: Omg you're right I misread the issue.

Comment (Marusyk) [reply src/Grok.Net/Grok.cs]: good point. thx

File: src/Grok.Net/GrokResult.cs [hunk 2]
```diff
@@ -9,5 +10,9 @@ public GrokResult(IList<GrokItem> grokItems)
             : base(grokItems ?? new List<GrokItem>())
         {
         }
+
+        public Dictionary<string, IEnumerable<object>> ToDictionary() => Items
```
Comment (Marusyk) [inline src/Grok.Net/GrokResult.cs:14]:
```suggestion
        public IReadOnlyDictionary<string, IEnumerable<object>> ToDictionary() => Items
```

**Review by Marusyk — APPROVED** [review]
LGTM 👍Thanks a lot

**Pull request merged**